third_party/intel/lib/TritonRaiseBlockPointer/TritonRaiseBlockPointer.cpp

```diff
@@ -237,6 +585,33 @@ struct TritonRaiseBlockPointer
 
     ptrMap.map(result, mapped);
 
+    // AddPtrOps that have been rewritten and no longer used in the code must be
+    // removed in the pass to avoid type matching issue.
+    cleanUp.push_back(op);
+
+    return success();
+  }
+
+  LogicalResult visitOperandMakeTensorPtr(triton::MakeTensorPtrOp makeTPtrOp,
+                                          PtrState &state, const Location loc,
+                                          OpBuilder &builder,
+                                          bool addedByPass = false) {
+    assert(state.isEmpty() && "state is a return argument");
+    state.source = makeTPtrOp.getBase();
+
+    auto resType = cast<triton::PointerType>(makeTPtrOp.getResult().getType());
+    auto pointeeType = cast<ShapedType>(resType.getPointeeType());
+    auto shape = pointeeType.getShape();
+
+    for (int64_t i = 0; i < pointeeType.getRank(); i++) {
+      state.sizes.push_back(shape[i]);
+
+      state.strides = makeTPtrOp.getStrides();
+      state.offsets = makeTPtrOp.getOffsets();
+      state.shape = makeTPtrOp.getShape();
+    }
```

Comment:
```suggestion
    state.sizes = SmallVector<int>(shape);

    state.strides = makeTPtrOp.getStrides();
    state.offsets = makeTPtrOp.getOffsets();
    state.shape = makeTPtrOp.getShape();
```

I think the 3 assignments can be out of the loop, for the sizes, not so sure due to typeconv